lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -74,11 +74,40 @@ static ParseResult
 parseOperation(OpAsmParser &parser,
```

Comment:
I think it would be better to just have a specialization for the parser of BufferOp now that we've diverged it's representation from the remainder of the operations.